Signal/src/AppDelegate.m

```diff
@@ -1147,6 +1147,18 @@ - (void)checkIfAppIsReady
     [[OWSMessageReceiver sharedInstance] handleAnyUnprocessedEnvelopesAsync];
     [[OWSBatchMessageProcessor sharedInstance] handleAnyUnprocessedEnvelopesAsync];
 
+    if (!Environment.preferences.hasGeneratedThumbnails) {
+        [OWSPrimaryStorage.sharedManager.newDatabaseConnection
```

Comment:
Generating thumbnails depends on the size of the media, but in my measuring is about 10 per second on an old iPhone5. We don't need to block launch of the app for this.

If the user goes straight to their gallery after first launch, it's possible they'll see missing thumbnails.

We could build a more robust approach, but it didn't seem very valuable as it's a fleeting problem that will solve itself for most people in a minute or two.